apps/dot_ext/loggers.py

```diff
@@ -0,0 +1,203 @@
+                                  "auth_app_id": str(application.id),
+                                  "auth_app_name": str(application.name),
+                                  "auth_client_id": str(application.client_id),
+                                  "auth_pkce_method": str(auth_pkce_method),
+                                  }
+                set_session_auth_flow_trace(request, auth_flow_dict)
+            except Application.DoesNotExist:
+                # Clear values in session. Set to empty value to denote not found.
+                auth_flow_dict = {"auth_uuid": new_auth_uuid,
+                                  "auth_app_id": "",
+                                  "auth_app_name": "",
+                                  "auth_client_id": "",
+                                  "auth_pkce_method": "",
+                                  }
+                set_session_auth_flow_trace(request, auth_flow_dict)
+                application = None
+
+        if application:
+            # Create and return AuthFlowUuid instance for tracking.
+            auth_flow_uuid = AuthFlowUuid.objects.create(auth_uuid=new_auth_uuid,
+                                                         client_id=application.client_id,
+                                                         auth_pkce_method=auth_pkce_method)
+
+    return auth_flow_uuid
+
+
+def get_session_auth_flow_trace(request):
+    '''
+    Function to get auth flow related items from the
+    session.
+
+    Returns a auth_flow_dict type DICT of values for logging.
+    '''
+    if request:
+        auth_flow_dict = {}
+        for k in SESSION_AUTH_FLOW_TRACE_KEYS:
+            auth_flow_dict[k] = request.session.get(k, None)
+        return auth_flow_dict
+    else:
+        return {}
+
+
+def set_session_auth_flow_trace(request, auth_flow_dict):
+    '''
+    Function to set auth flow related items in the
+    session from a dictionary.
+    '''
+    if request:
+        for k in SESSION_AUTH_FLOW_TRACE_KEYS:
+            request.session[k] = auth_flow_dict.get(k, None)
+
+
+def set_session_values_from_auth_flow_uuid(request, auth_flow_uuid):
+    '''
+    Function to set auth flow related items in the
+    session given an AuthFlowUuid instance.
+    '''
+    application = None
+    if auth_flow_uuid:
+        request.session['auth_uuid'] = str(auth_flow_uuid.auth_uuid)
+        request.session['auth_pkce_method'] = str(auth_flow_uuid.auth_pkce_method)
+
+        # Get the application.
+        Application = get_application_model()
+        try:
+            application = Application.objects.get(client_id=auth_flow_uuid.client_id)
+        except Application.DoesNotExist:
+            pass
+
+    if application:
+        # Set values in session.
+        request.session['auth_app_id'] = str(application.id)
+        request.session['auth_app_name'] = str(application.name)
+        request.session['auth_client_id'] = str(application.client_id)
```

Comment:
Technically you can move these inside the try statement and get rid of the if because if an application does not exist this will never complete anyway.